server/etcdserver/v3_server.go

```diff
@@ -315,7 +333,13 @@ func (s *EtcdServer) LeaseRenew(ctx context.Context, id lease.LeaseID) (int64, e
 }
 
 func (s *EtcdServer) LeaseTimeToLive(ctx context.Context, r *pb.LeaseTimeToLiveRequest) (*pb.LeaseTimeToLiveResponse, error) {
-	if s.Leader() == s.ID() {
+	cctx, cancel := context.WithTimeout(ctx, s.Cfg.ReqTimeout())
```

Comment:
This is incorrect place to apply timeout. As variable name suggest `s.Cfg.ReqTimeout()` is a timeout for requests. In case of leases, this timeout is used when trying to forward request to leader. Applying this timeout to followers or waiting for apply index is incorrect usage of such timeout.

Reply:
Thanks for the comment, but does it make more sense to regard the `s.Cfg.ReqTimeout()` as the whole timeout? From the users/clients perspective, they don't care how etcd internally process the request. So the `s.Cfg.ReqTimeout()` should contain all times, including the `waitAppliedIndex` and `forward to leader`?

Reply:
The method `waitAppliedIndex` will only be called when the current member is the leader (please see line 339 and 340.  In other words, if current member isn't the leader, then logic (including the context/timeout) is exactly the same as previous implementation.

Reply:
Yes, however I'm worried about setting the incorrect request timeout on the leader for `waitAppliedIndex`.

Reply:
Yes, it's a valid comment. Previously the waitAppliedTimeout is 1 second, see [http.go#L35](https://github.com/etcd-io/etcd/blob/main/server/lease/leasehttp/http.go#L35).  It seems we need to follow the same timeout value (1s), and we can set a common variable to be share by both places.

Reply:
Done. I just set a separate [applyTimeout](https://github.com/ahrtr/etcd/blob/lease_renew_linearizable/server/etcdserver/v3_server.go#L51) for v3 lease, also added a unit test. 

My understanding is that we may need to remove the existing [v2 lease http file](https://github.com/etcd-io/etcd/blob/main/server/lease/leasehttp/http.go). The followers should use v3client to forward the lease requests to leader. This is definitely not in the scope of this PR.  cc @serathius @ptabor